### DISCUSSION TITLE
Error "Cannot access a closed Stream" when building WinUI app

### DISCUSSION THREAD

**Discussion by Juriyx** [unoplatform/uno#17169] [Q&A] [answered]
Hello. I build a WinUI app in Azure DevOps with `Debug` configuration and got such error on a build step:

```
Unhandled Exception: System.ObjectDisposedException: Cannot access a closed Stream.
   at System.IO.__Error.StreamIsClosed()
   at System.IO.MemoryStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.StreamWriter.Write(Char[] buffer, Int32 index, Int32 count)
   at System.IO.TextWriter.SyncTextWriter.WriteLine(String value)
   at System.Console.WriteLine(String value)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String commandLine)
   at Microsoft.Build.CommandLine.MSBuildApp.Main()
```

Below is the pipeline I use:
[Windows app distribution (Debug).txt](https://github.com/user-attachments/files/15854867/Windows.app.distribution.Debug.txt)

**Comment by jeromelaban** [comment]
Thanks for the report. Are you able to [provide a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html)? Could you also specify which step is causing this issue?

**Reply by Juriyx** [reply]
Yes, sure. We also got `MSBuild_*.failure.txt` log files
[WinUI build errors.zip](https://github.com/user-attachments/files/15876283/WinUI.build.errors.zip)

**Reply by jeromelaban** [reply]
Thanks. This is not an issue we've encountered before. It's likely a bug in msbuild, most likely due to parallelism. You may want to disable any parallelism options, such as `maximumCpuCount: true` and add `/m:1` to the msbuild command line.

**Reply by Juriyx** [reply]
I made target framework overriding by creating solution-config.props file with this content:
```
<Project>
  <PropertyGroup>
    <OverrideTargetFramework Condition="''!='hint: Windows App Sdk (WinUI)'">$(DotNetVersion)-windows10.0.22621</OverrideTargetFramework>
  </PropertyGroup>
</Project>
```
and used build steps from this [tutorial](https://platform.uno/docs/articles/uno-publishing-windows-packaged-signed.html).

In result, the build terminates and I don't see any errors in the pipeline log. Here is the binlog file:
[WinAppX64.build.binlog.zip](https://github.com/user-attachments/files/16091072/WinAppX64.build.binlog.zip)

**Reply by Juriyx** [reply]
@jeromelaban Could you answer my question above, please?

**Reply by jeromelaban** [reply]
Thanks for the update. I'm not able to reproduced this issue, and this is a .NET issue we have no control over. 

I'd suggest trying not using the msbuild task, but rather call msbuild directly from a powershell or cmd task, it may change the behavior.

**Reply by Juriyx** [reply]
Interesting tip) I'll give it a try and report back on the result. Thank you!

**Reply by Juriyx** [reply]
> I'd suggest trying not using the msbuild task, but rather call msbuild directly from a powershell or cmd task

Unfortunately, I haven't found the working way to call msbuild. It seems that it is not added to the system's PATH for some reason. What can be done regarding this?

**Reply by jeromelaban** [reply, marked as answer]
You need to find msbuild using vswhere: https://github.com/unoplatform/uno/blob/b9ecb0837c64333daaa4b272913786d3f0928989/build/test-scripts/run-netcore-mobile-template-tests.ps1#L22

**Reply by Juriyx** [reply]
Thanks, @jeromelaban! After this task implementation, we were able to see errors and fix the build

**Reply by jeromelaban** [reply]
Thanks for the update. What change did you have to make?
